### PR TITLE
WELD-2200 replace MetaDataImpl.

### PR DESCRIPTION
**Pull request by tremes**

### CODE REVIEW

**Comment by tremes** [comment]
retest this please

File: impl/src/main/java/org/jboss/weld/metadata/MetadataImpl.java [hunk 1]
```diff
@@ -1,12 +1,33 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package org.jboss.weld.metadata;
 
 import org.jboss.weld.bootstrap.spi.Metadata;
 
+@Deprecated
```
Comment (mkouba) [inline impl/src/main/java/org/jboss/weld/metadata/MetadataImpl.java]: Should we add `@deprecated` javadoc tag referencing MetadataImpl from API?

Comment (tremes) [inline impl/src/main/java/org/jboss/weld/metadata/MetadataImpl.java]: Could be.

**Comment by mkouba** [comment]
Merged, thanks!

**Pull request closed without merging**